### DISCUSSION TITLE
Collections as home sections

### DISCUSSION THREAD

**Discussion by lostb1t** [jellyfin/jellyfin-meta#83] [Proposals]
Now that its possible to sort collections by added date, things like trending, popular, top 100s etc are a fun way to provide my users with media suggestions. (see my plugin for an example of external collections: https://github.com/lostb1t/jellyfin-plugin-collection-import)

The next step would be to allow collections as home sections. The media items within a collection will be shown similar to recently added etc on home. Making the home screen more on par with apps like plex or netflix. This will be optional so a user does not have to use it.

From a technical perspective this means

1. Adding collections to the section options in the web settings
2. Add server side validation for section collection options
3. Implement the collection section rows on home on web. The api calls is just an item query with the collection id as parent so no server side changes needed.

It’s a fairly unobtrusive code change, at least for web. Other clients can implement it as they see fit as there are no breaking changes server side.

**Comment by thornbill** [comment]
- We will need to add a new home section type in the server: https://github.com/jellyfin/jellyfin/blob/1272bb9a8420a92329cdd36898b4b0240018327f/Jellyfin.Data/Enums/HomeSectionType.cs#L6
- In the past some clients have crashed when getting an unexpected type... I think that is no longer the case but we should verify. 

Related feature requests:
- https://features.jellyfin.org/posts/1439/custom-carousel-options-on-home-screen
- https://features.jellyfin.org/posts/2049/collection-necessities
- https://features.jellyfin.org/posts/1986/new-or-improved-home-secreen-sections

**Reply by lostb1t** [reply]
for testing purposes, which clients are we talking about out here? the mobile versions are just wrappers so those aren't an issue.

**Reply by jimdogx** [reply]
As an FYI, Roku clients should be good with an unexpected type.  It's currently logged and ignored.

**Reply by thornbill** [reply]
The other native clients:
- Android TV
- Swiftfin

**Comment by lostb1t** [comment]
created a quick and dirty POC:

Im using customprefs to store the collection id as i don't want to edit the db scheme because of the db migration.

server: https://github.com/lostb1t/jellyfin/tree/feature/collectionsection
web: https://github.com/lostb1t/jellyfin-web/tree/feature/collectionsection

<img width="981" alt="Screenshot 2024-12-01 at 09 05 19" src="https://github.com/user-attachments/assets/cfdcb275-909d-419e-a50d-3aa25c21ede2">
<img width="1168" alt="Screenshot 2024-12-01 at 09 04 51" src="https://github.com/user-attachments/assets/466bccb0-cd5d-4880-afa2-ff511cf18f69">

**Reply by johnpc** [reply]
This is very exciting! Do you have any ideas about what a productionalized implementation of this would look like? I am also motivated to realize this idea (I actually created https://github.com/jellyfin/jellyfin-meta/discussions/93 before I realized this discussion already existed).

I am willing to volunteer time to implement this feature on the server and / or on whatever client once the server is ready, including whatever refactor the maintainers want to see with regard to how Home Screen Sections work presently.

If we can align on a design, I am happy to help implement it!

**Comment by thornbill** [comment]
@lostb1t I believe you said in chat that you are no longer working on this. If that is the case perhaps we should close this in favor of https://github.com/jellyfin/jellyfin-meta/discussions/93 ?

**Comment by lostb1t** [comment]
Closing in favor of #93
